rotkehlchen/history/events/structures/eth2.py

```diff
@@ -207,9 +213,18 @@ def process(
             accounting: 'AccountingPot',
             events_iterator: Iterator['AccountingEventMixin'],  # pylint: disable=unused-argument
     ) -> int:
+        # this is to avoid a circular import.
+        from rotkehlchen.db.eth2 import DBEth2
+
+        balances, _, _ = DBEth2(accounting.database).process_validators_balances_and_pnl(
+            from_ts=Timestamp(0),
+            to_ts=Timestamp(self.get_timestamp_in_sec() - 1),  # get the balance just before this event.  # noqa: E501
+            validator_indices={self.validator_index},
+        )
+        max_effective_balance = MAX_EFFECTIVE_BALANCE if list(balances[self.validator_index].values())[-1] > MIN_EFFECTIVE_BALANCE else MIN_EFFECTIVE_BALANCE  # noqa: E501
         profit_amount = self.amount
-        if self.amount >= 32:
-            profit_amount = self.amount - 32
+        if self.amount >= max_effective_balance:
+            profit_amount = self.amount - max_effective_balance
```

Comment:
This should have an if statement  where you check if `self.validator_index` is an accumulating or distributing validator. If it is a distributing you want to keep the old logic here and avoid the process_validators thing